src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -34,14 +34,23 @@ class DelegatingLoader extends BaseDelegatingLoader
     /**
      * Constructor.
      *
-     * @param ControllerNameParser    $parser   A ControllerNameParser instance
-     * @param LoggerInterface         $logger   A LoggerInterface instance
-     * @param LoaderResolverInterface $resolver A LoaderResolverInterface instance
+     * Ability to pass a LoggerInterface instance as second argument will be removed in 3.0.
+     *
+     * @param ControllerNameParser                    $parser   A ControllerNameParser instance
+     * @param LoggerInterface|LoaderResolverInterface $logger   A LoggerInterface instance
+     * @param LoaderResolverInterface|null            $resolver A LoaderResolverInterface instance
      */
-    public function __construct(ControllerNameParser $parser, LoggerInterface $logger = null, LoaderResolverInterface $resolver)
+    public function __construct(ControllerNameParser $parser, $logger = null, $resolver = null)
```

Comment:
The variable names and the docblock should reflect the behaviour in Symfony 3.0.